apps/context.go

```diff
@@ -4,23 +4,56 @@ import (
 	"github.com/mattermost/mattermost-server/v5/model"
 )
 
+// Context is included in CallRequest and provides App with information about
+// Mattermost environment (configuration, authentication), and the context of
+// the user agent (current channel, etc.)
+//
+// To help reduce the need to go back to Mattermost REST API, ExpandedContext
+// can be included by adding a corresponding Expand attribute to the originating
+// Call.
 type Context struct {
-	AppID             AppID             `json:"app_id"`
-	Location          Location          `json:"location,omitempty"`
-	Subject           Subject           `json:"subject,omitempty"`
-	BotUserID         string            `json:"bot_user_id,omitempty"`
-	ActingUserID      string            `json:"acting_user_id,omitempty"`
-	UserID            string            `json:"user_id,omitempty"`
-	TeamID            string            `json:"team_id"`
-	ChannelID         string            `json:"channel_id,omitempty"`
-	PostID            string            `json:"post_id,omitempty"`
-	RootPostID        string            `json:"root_post_id,omitempty"`
-	Props             map[string]string `json:"props,omitempty"`
-	MattermostSiteURL string            `json:"mattermost_site_url"`
-	UserAgent         string            `json:"user_agent,omitempty"`
+	// AppID is used for handling CallRequest internally, and is always filled
+	// in.
+	AppID AppID `json:"app_id"`
+
+	// Fully qualified original Location of the user action (if applicable),
+	// e.g. "/command/helloworld/send" or "/channel_header/send".
+	Location Location `json:"location,omitempty"`
+
+	// Subject is a subject of notification, if the call originated from a
+	// subscription.
+	Subject Subject `json:"subject,omitempty"`
+
+	// BotUserID of the App.
+	BotUserID string `json:"bot_user_id,omitempty"`
+
+	// ActingUserID is primarily (or exclusively?) for calls originating from
+	// user submissions.
+	ActingUserID string `json:"acting_user_id,omitempty"`
```

Comment:
I would go for exclusively.
For notifications I would use UserID, since the one generating the call is Mattermost, not the user, and the user may not be even active (e.g. user joined channel when someone else invite you to a channel).